src/provider/element-templates/Validator.js

```diff
@@ -101,7 +103,26 @@ export class Validator {
       }
     }
 
-    // (3) JSON schema compliance
+    // (3) elementType validation
+    if (template.elementType && template.appliesTo) {
+
+      const elementType = template.elementType.value,
+            appliesTo = template.appliesTo;
+
+      // (3.1) template can be applied to elementType
+      if (!appliesTo.find(type => isType(elementType, type))) {
+        return this._logError(`template does not apply to requested element type <${ elementType }>`, template);
+      }
+
+      // (3.2) template only applies to same type of element
+      for (const sourceType of appliesTo) {
+        if (!canMorph(elementType, sourceType)) {
+          return this._logError(`can not morph <${sourceType}> into <${elementType}>`, template);
+        }
+      }
+    }
+
+    // (4) JSON schema compliance
```

Comment:
I see quite the same implementation twice. Maybe we can implement it once here, e.g. as private method `_validateElementType` of the `elementTemplate` validator, and re-use it in the other (which is already extending the first one)?

Reply:
I did not see that it get's extended! That makes sense, it is the same test after all